tests/lib/test_workflow.py

```diff
@@ -0,0 +1,95 @@
+# pylint: disable=missing-class-docstring disable=missing-function-docstring disable=missing-module-docstring
+
+from unittest import TestCase
+from unittest.mock import call, patch
+
+import nestor_api.lib.workflow as workflow
+
+
+class TestWorkflow(TestCase):
+    @patch("nestor_api.lib.workflow.git")
+    def test_compare_unequal_step_hashes(self, git_mock):
+        """Should return False when step hashes are unequal."""
+        git_mock.get_last_commit_hash.side_effect = ["hash-1", "hash-2"]
+        result = workflow.compare_step_hashes("path_to/app_dir", "step-1", "step-2")
+        self.assertEqual(git_mock.branch.call_count, 2)
+        self.assertEqual(git_mock.get_last_commit_hash.call_count, 2)
+        git_mock.branch.assert_has_calls(
+            [call("path_to/app_dir", "step-1"), call("path_to/app_dir", "step-2")]
+        )
+        git_mock.get_last_commit_hash.assert_has_calls(
+            [call("path_to/app_dir"), call("path_to/app_dir")]
+        )
+        self.assertFalse(result)
```

Comment:
Would be more readable IMO but up to you.
```suggestion
        git_mock.get_last_commit_hash.side_effect = ["hash-1", "hash-2"]
        
        result = workflow.compare_step_hashes("path_to/app_dir", "step-1", "step-2")
        
        self.assertEqual(git_mock.branch.call_count, 2)
        self.assertEqual(git_mock.get_last_commit_hash.call_count, 2)
        git_mock.branch.assert_has_calls(
            [call("path_to/app_dir", "step-1"), call("path_to/app_dir", "step-2")]
        )
        git_mock.get_last_commit_hash.assert_has_calls(
            [call("path_to/app_dir"), call("path_to/app_dir")]
        )
        self.assertFalse(result)
```